tests/pytests/test_aggregate_count.py

```diff
@@ -0,0 +1,695 @@
+    else:
+        shard = res[1][1][0][13]
+        coord = res[1][3][11]
+        shard_RP = [item[1] for item in shard]
+        coord_RP = [item[1] for item in coord]
+        return [shard_RP, coord_RP]
+
+
+def _get_standalone_RP_profile(env, res) -> list:
+    if isinstance(res, dict):
+        profile = res['Profile']['Shards'][0]['Result processors profile']
+        return [item['Type'] for item in profile]
+    else:
+        profile = res[1][1][0][13]
+        return [item[1] for item in profile]
+
+
+def _translate_query_to_profile_query(query) -> list:
+    profile = ['FT.PROFILE']
+    profile.append(query[1])        # index name
+    profile.append(query[0][3:])    # command
+    profile.append('QUERY')
+    profile.extend(query[2:])       # query
+    return profile
+
+
+def _test_limit00(protocol):
+    env = Env(protocol=protocol)
+    docs = 2265
+    _setup_index_and_data(env, docs)
+
+    for on_timeout_policy in ['return', 'fail']:
+        config_cmd = ['CONFIG', 'SET', 'search-on-timeout', on_timeout_policy]
+        verify_command_OK_on_all_shards(env, *config_cmd)
+        queries_and_results = [
+            (['FT.AGGREGATE', 'idx', '*', 'WITHCOUNT', 'LIMIT', 0, 0], docs),
+            (['FT.AGGREGATE', 'idx', '*', 'WITHOUTCOUNT', 'LIMIT', 0, 0], ANY),
+            # WITHOUTCOUNT is implied by default
+            (['FT.AGGREGATE', 'idx', '*', 'LIMIT', 0, 0], ANY),
+        ]
+
+        for query, expected_results in queries_and_results:
+            cmd=' '.join(str(x) for x in query)
+            for dialect in [1, 2, 3, 4]:
+                config_cmd = ['CONFIG', 'SET', 'search-default-dialect', dialect]
+                verify_command_OK_on_all_shards(env, *config_cmd)
+                res = env.cmd(*query)
+                total_results = _get_total_results(res)
+                results = _get_results(res)
+
+                # Verify results
+                env.assertEqual(
+                    total_results, expected_results,
+                    message=f'{cmd}: total_results != expected. Dialect: {dialect}')
+                env.assertEqual(
+                    len(results), 0,
+                    message=f'{cmd}: len(results) != 0. Dialect: {dialect}')
+
+
+def test_limit00_resp3():
+    _test_limit00(3)
+
+
+def test_limit00_resp2():
+    _test_limit00(2)
+
+
+def _test_withcount(protocol):
+    env = Env(protocol=protocol)
+    docs = 2265
+    _setup_index_and_data(env, docs)
+
+    queries_and_results = [
+        # query, total_results, length of results
+
+        # WITHCOUNT
+        # No sorter, no limit, returns all results
+        (['FT.AGGREGATE', 'idx', '*', 'WITHCOUNT'], docs, docs),
+
+        # WITHCOUNT + LIMIT
+        # No sorter, limit results
+        # total_results = number of documents matching the query up to the LIMIT
+        # length of results = min(total_results, LIMIT)
+        (['FT.AGGREGATE', 'idx', '*', 'WITHCOUNT', 'LIMIT', 0, 50], docs, 50),
```

Comment:
Can we also check the actual number of rows returned from the shard to the coord?
Need to make sure that the rows the shards are sending sums up to the length of results the coordinator is sending to the client

Reply:
Done. commit efd0a5e1d
Tested in `_test_profile`